### PR TITLE
Remove deprecated `prepare_data_per_node` in `Trainer`

### PR DESCRIPTION
**Pull request by yhl48**
## What does this PR do?
Part of https://github.com/PyTorchLightning/pytorch-lightning/issues/12521

### Does your PR introduce any breaking changes? If yes, please list them.
Remove deprecated prepare_data_per_node in Trainer

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,9 +61,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 
 - Removed the deprecated `progress_bar_refresh_rate` argument from the `Trainer` constructor ([#12514](https://github.com/PyTorchLightning/pytorch-lightning/pull/12514))
-
-
--
+- Remove deprecated `prepare_data_per_node` flag from `Trainer` ([#12536](https://github.com/PyTorchLightning/pytorch-lightning/pull/12536))
```
Comment (Borda) [inline CHANGELOG.md:64]:
```suggestion


- Remove deprecated `prepare_data_per_node` flag from `Trainer` ([#12536](https://github.com/PyTorchLightning/pytorch-lightning/pull/12536))
```

**Review by Borda — APPROVED** [review]

**Comment by rohitgr7** [comment]
> Not sure if the doc here needs changing? Would appreciate any advice here, thanks.
> 
> https://github.com/PyTorchLightning/pytorch-lightning/blob/cf65ca25b0230df89eec77c218c5e928c8141840/pytorch_lightning/core/hooks.py#L364

this should be good:
```
In DDP ``prepare_data`` can be called in two ways (using ``prepare_data_per_node``):
```
since there is already a reference for it right below :)

**Review by rohitgr7 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed the deprecated `progress_bar_refresh_rate` argument from the `Trainer` constructor ([#12514](https://github.com/PyTorchLightning/pytorch-lightning/pull/12514))
 
 
--
+- Remove deprecated `prepare_data_per_node` flag from `Trainer` ([#12536](https://github.com/PyTorchLightning/pytorch-lightning/pull/12536))
```
Comment (kaushikb11) [inline CHANGELOG.md:66]:
```suggestion
- Removed the  deprecated `prepare_data_per_node` argument from the `Trainer` constructor ([#12536](https://github.com/PyTorchLightning/pytorch-lightning/pull/12536))
```
For consistency with other changelog entries

**Review by kaushikb11 — APPROVED** [review]

File: pytorch_lightning/core/hooks.py [hunk 3]
```diff
@@ -361,21 +361,27 @@ def prepare_data(self):
                 self.split = data_split
                 self.some_state = some_other_state()
 
-        In DDP ``prepare_data`` can be called in two ways (using Trainer(prepare_data_per_node)):
+        In DDP ``prepare_data`` can be called in two ways
```
Comment (ananthsub) [inline pytorch_lightning/core/hooks.py:364]:
```suggestion
        In a distributed environment, ``prepare_data`` can be called in two ways
```

**Review by ananthsub — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
